### ISSUE TITLE
Table not honouring container width

### ISSUE THREAD

**Issue by phllipo** [freenowtech/wave#60]
While implementing a table using the table components I noticed that the table is not honouring the container width.
<img width="1124" alt="current_state" src="https://user-images.githubusercontent.com/9133431/116530652-8e632380-a8de-11eb-8476-e37d83eb66d3.png">

After some research I found that adding `table-layout: fixed` fixes the problem.
<img width="1068" alt="fixed_no_widths" src="https://user-images.githubusercontent.com/9133431/116530970-ebf77000-a8de-11eb-94f2-9ec983008e0d.png">

That also gives the opportunity to have custom width columns, by adding width to the `<th>` components.
<img width="1083" alt="fixed_custom_widths" src="https://user-images.githubusercontent.com/9133431/116531046-0af60200-a8df-11eb-9acb-85908fb95302.png">

Do you think adding `table-layout: fixed` to the table has any downsides?
And if that would be added should the header component also have a width prop?

**Comment by rafael-sepeda** [comment]
@phllipo According to https://a11y-101.com/development/tables tables support multiple lines. But at the same time in [wave](https://wave.free-now.com/components/Table) we documented that:
> Keep in mind, that the smallest row size will not work when you have multiple lines in one cells.

Having this in mind I would propose to have `table-layout: fixed` optional (if we need it), but in most cases break the text into multiple lines. We only need to see how this works with the `rowSize : small `

For the header component I would not expect to have a `fixed` property

**Comment by phllipo** [comment]
@rafael-sepeda yes in theory multiple lines are fine with tables, but as far as I see we have `white-space: nowrap` and a fixed height on the TableCell which goes against multiple lines.
Also without the `table-layout: fixed` I think you can't make columns have a custom width.

Never mind the width prop for the header, just found that's already working.

**Comment by rafael-sepeda** [comment]
@phllipo Ok, I see. Then I'm fine with what you're proposing. In case if we will see that we need to still have multiline in some use cases we can still override nowrap settings in place (with `word-wrap: break-word !important` `white-space: normal !important` or other), right?

**Comment by phllipo** [comment]
@rafael-sepeda yes overriding is not a problem. And multiline will also work fine with `table-layout: fixed`.

**Comment by duranmla** [comment]
I am a bit confused with this one @phllipo I have tried to go ahead and reproduce the scenario but `table-layout: fixed` won't do the fix. https://codesandbox.io/s/cocky-currying-3gs8o maybe I am missing something?

**Comment by duranmla** [comment]
I have updated the [codesandbox](https://codesandbox.io/s/cocky-currying-3gs8o), it seems with `width: 100%` combined with `table-layout: fixed` or `table-layout: auto` has two different valid outputs that should respect the container width. Check https://developer.mozilla.org/en-US/docs/Web/CSS/table-layout

My concern here is that it feels that using `table-layout: fixed` will enforce the usage of defined width on headers as @phllipo commented. Maybe we can just keep the current behavior and people that need more specific behavior will use `style={}` prop?. This will free us from trying to make abstractions that are somehow complex to explain _(as will involve different components from the table)_
